src/components/accordion/CdrAccordion.vue

```diff
@@ -1,42 +1,187 @@
+<script setup>
+import {
+  useCssModule, computed, watch, onMounted, ref, inject,
+} from 'vue';
+import IconCaretDown from '../icon/comps/caret-down';
+import { modifyClassName } from '../../utils/buildClass';
+import mapClasses from '../../utils/mapClasses';
+
+
+const props = defineProps({
+  /**
+   * The unique id of an accordion.
+   */
+  id: {
+    type: String,
+    required: true,
+  },
+  /**
+   * Toggle this value to open/close the accordion.
+   */
+  opened: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Sets a compact style.
+   */
+  compact: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Sets a border-aligned style.
+   */
+  borderAligned: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Sets the heading level
+   */
+  level: {
+    type: [String, Number],
+    required: true,
+  },
+  /**
+   * Toggles content spacing (padding)
+   */
+  contentSpacing: {
+      type: Boolean,
+      default: true,
+    },
+  label: {
+    type: String,
+  },
+});
+const emit = defineEmits(['accordion-toggle']);
+const unwrap = inject('unwrap');
+const style = useCssModule();
+const accordionContentEl = ref(null);
+const focused = ref(false);
+const maxHeight = ref(props.opened
+  ? 'none'
+  : 0);
+const headingTag = `h${props.level}`;
+const labelClass = 'cdr-accordion__label';
+const baseClass = 'cdr-accordion';
+const iconClass = 'cdr-accordion__icon';
+const containerClass = 'cdr-accordion__content-container';
+const contentClass = 'cdr-accordion__content';
+
+const isUnwrapped = computed(() => unwrap && unwrap.isUnwrapped);
+const headingContent = computed(() => (isUnwrapped.value
```

Comment:
I dislike how much additional logic is required to facilitate this unwrapped state. I would like to revisit this in a future update - maybe drop a comment in here to remind us?